sdk/dapp-kit/src/hooks/wallet/useSwitchAccount.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { WalletAccount } from '@mysten/wallet-standard';
+import type { UseMutationOptions } from '@tanstack/react-query';
+import { useMutation } from '@tanstack/react-query';
+import { useWalletContext } from '../../components/WalletProvider.js';
+import { walletMutationKeys } from '../../constants/walletMutationKeys.js';
+import { WalletAccountNotFoundError, WalletNotConnectedError } from '../../errors/walletErrors.js';
+
+type SwitchAccountArgs = {
+	account: WalletAccount;
+};
+
+type SwitchAccountResult = void;
+
+type UseSwitchAccountMutationOptions = Omit<
+	UseMutationOptions<SwitchAccountResult, Error, SwitchAccountArgs, unknown>,
+	'mutationFn'
+>;
+
+/**
+ * Mutation hook for switching to a specific wallet account.
+ */
+export function useSwitchAccount({
```

Comment:
Talked about this with @Jordan-Mysten a bit, and we're going to make this a mutation hook despite it not having any async actions to A). provide a more consistent interface for hook consumers and B). have these errors be thrown in a manageable manner for consumers